src/main/java/cn/nukkit/block/BlockNoteblock.java

```diff
@@ -219,15 +211,30 @@ public boolean onActivate(Item item, Player player) {
     @Override
     public int onUpdate(int type) {
         if (type == Level.BLOCK_UPDATE_REDSTONE) {
-            if (this.getLevel().isBlockPowered(this)) this.emitSound();
+            BlockEntityMusic blockEntity = this.getBlockEntity();
+            if (this.getLevel().isBlockPowered(this)) {
+                if (!blockEntity.isPowered()) {
+                    this.emitSound();
+                }
+                blockEntity.setPowered(true);
+            } else {
+                blockEntity.setPowered(false);
+            }
+        }
+        return super.onUpdate(type);
+    }
+
+    private BlockEntityMusic getBlockEntity() {
+        BlockEntity blockEntity = this.getLevel().getBlockEntity(this);
+        if (blockEntity instanceof BlockEntityMusic) {
+            return (BlockEntityMusic) blockEntity;
         }
-        return 0;
+        return this.createBlockEntity();
```

Comment:
This should return null if it cannot find one otherwise it will keep creating new block entities.